elastic/kibana
# [Canvas] Disable Edit in Lens action for legacy savedVisualization Canvas function

## Summary

Removes the Edit in Lens action from the context menu for legacy `savedVisualization` Canvas function. Prior to this change, attempts to convert a `savedVisualization` to a Lens visualization would crash Canvas due to an unhandled error.

It may be possible in a future PR to migrate the `savedVisualization` to an `embeddable` function when the user chooses "Convert to Lens". 

Fixes #152662 

To test this PR: 
1. Add Sample data web logs. 
2. Create a new legacy goal visualization and save to library.
3. Add the goal visualization to a Canvas workpad.
4. Notice how the expression for the element uses the `embeddable` function.
5. Make a copy of the element on the workpad.
6. Find the saved object id for the visualization (you can either locate it in the Stack Management-Saved Objects or decode the base64 in the config for the element (e.g. `JSON.parse(atob("<base64 string>"))`. 
7. Change the expression for one of the elements to `savedVisualization id="<saved object id>"` and be sure the visualization renders correctly.
8. Open the context menu for the element and notice the "Convert to Lens" option is not present.
9. Click the "Edit visualization" option in the context menu.
10. In the Visualization editor, choose the "Edit in Lens" link at the top of the page.
11. In the Lens editor, click "Replace in Canvas". 
12. Notice a toast error specifies that the legacy `savedVisualization` function can not be converted to Lens.
13. Select the element that uses the `embeddable` function and try steps 8-11 above. It should be possible for this element to "Convert to Lens".




https://user-images.githubusercontent.com/1638483/230918702-356ee6b6-da30-4a56-a206-9588a6080032.mov
